File: api/Models/Department.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Models
{
    [Table("Departement", Schema = "Faculty")]
    public class Department
    {
        [Key]
        public int Dept_ID { get; set; }

        [Required]
        [StringLength(100)]
        public string Dept_Name { get; set; } = string.Empty;

        [Required]
        public int Doc_ID { get; set; }

        [Required]
        public DateTime Establish_Date { get; set; }

        [StringLength(50)]
        public string? Location { get; set; }

        [StringLength(200)]
        public string? Dept_Description { get; set; }

        [StringLength(20)]
        public string? Dept_Code { get; set; }

        // Navigation properties
        [ForeignKey("Doc_ID")]
        public virtual Doctor? HeadDoctor { get; set; }

        public virtual ICollection<Doctor> Doctors { get; set; } = new List<Doctor>();
        public virtual ICollection<Student> Students { get; set; } = new List<Student>();
        public virtual ICollection<Course> Courses { get; set; } = new List<Course>();
    }
}

File: frontend/types/Department.ts
// Department interface matching Faculty_System database schema
export interface Department {
  Dept_ID: number;
  Dept_Name: string;
  Doc_ID: number; // Head of department (Doctor ID)
  Establish_Date: string;
  Location?: string;
  Dept_Description?: string;
  Dept_Code?: string;
  // Related data
  headDoctor?: {
    Doc_ID: number;
    FName: string;
    LName: string;
    Email: string;
  };
}

export interface CreateDepartmentRequest {
  Dept_ID: number;
  Dept_Name: string;
  Doc_ID: number;
  Establish_Date: string;
  Location?: string;
  Dept_Description?: string;
  Dept_Code?: string;
}

export interface UpdateDepartmentRequest extends Partial<CreateDepartmentRequest> {}
